### PR TITLE
feat: persist normalized watch events to SQLite

### LINKED ISSUE #24: feat: persist normalized watch events to SQLite, make provider zips import-only
## Problem

Raw event history is not persisted in a reusable normalized form after ingestion. The zip is still a runtime dependency because:

- Ordinary recommendations run from cached artifacts (`watch_index.json`, `taste_profile.txt`)
- Abandoned-watch queries re-read the original provider zips at runtime

If we persist normalized watch events after import, the zip becomes optional after ingestion.

## Options considered

| Option | Verdict |
|--------|---------|
| CSV | Workable but weak for dedup, provenance, schema changes, and querying |
| JSON / JSONL | Good minimal first step; you end up hand-rolling indexing and dedup |
| SQLite | Best long-term fit — single local file, structured schema, dedup, incremental imports, no server |
| Rebuild into `watch_index.json` only | Insufficient — not enough detail for abandoned-watch and future event-based features |

## Proposed design

Use a small local SQLite database in `recommender/cache/` or `data/`.

### Schema

**`watch_events`**
- `provider`
- `title`
- `content_type`
- `series_name`
- `watched_duration_seconds`
- `timestamp_utc`
- `profile`
- `import_id`
- `source_hash` (dedup key)

**`imports`**
- `id`
- `provider`
- `source_path`
- `source_sha256`
- `imported_at`

### Behavior

- `./recommend setup --ingest-only` validates and imports, writing normalized events to SQLite
- `watch_index.json` and `taste_profile.txt` are derived from the normalized event store, not from re-parsing zips
- Runtime (`./recommend`, web UI) reads events from SQLite — provider zips not required
- Re-import defaults to one active import per provider; new import replaces old provider events unless user requests a merge

## Recommended incremental path

1. Persist normalized events to JSONL after import (smallest change)
2. Load `ctx.events` from that persisted store at runtime
3. Stop re-parsing provider zips at runtime
4. Migrate store from JSONL to SQLite in a follow-up

**SQLite is the best long-term design. JSONL is the best incremental transition.**

## Related

- #23 — multi-zip support per provider (that feature becomes easier once the store is normalized)

### PR DESCRIPTION
**Pull request by abhichandra21**
## Summary

Closes #24.

Provider zip files are now import-only. After running \`./recommend setup\` or \`./recommend setup --ingest-only\`, the runtime (web UI, CLI) loads watch history from \`data/streamline.db\` instead of re-parsing zip files on every startup.

- **New \`recommender/event_store.py\`** — thin SQLite API: \`init_db\`, \`replace_provider_events\` (atomic replace-on-reimport with source_hash dedup), \`load_events\`, \`get_import_info\`
- **\`RecommendContext.events\`** is now a lazy \`@property\` — only loads from SQLite when an abandoned-watch query actually needs it; ordinary recommendations are unaffected
- **\`setup.py\`** — default setup writes to SQLite after validation, loads back as source of truth (dedup reflected in derived artifacts); \`--refresh-data\`/\`--refresh-profile\` read from SQLite instead of re-parsing zips; \`--ingest-only\` validates and persists
- **\`main.py\` / \`web.py\`** — \`load_context()\` / \`_build_context()\` no longer touch provider zips; use \`_events_loader\` lazy closure instead
- **\`/status\`** — always returns \`"ok"\`; adds \`event_store_ready\`, \`event_store_import_count\`, \`event_store_path\` (event store absence is informational, not a health gate)
- **\`/logs\`** — supports \`?file=app|web\` to switch between log files; new \`POST /logs/clear\` endpoint (CSRF-protected) truncates the selected file

## Test plan

- [ ] \`venv/bin/python -m pytest tests/ -q\` → 208/208 pass
- [ ] \`./recommend setup\` on real data → check \`data/streamline.db\` created, recommendations work
- [ ] \`./recommend setup --ingest-only\` → check import row in SQLite, no derived artifacts built
- [ ] \`./recommend setup --refresh-data\` → verify no zip files needed, loads from SQLite
- [ ] \`./recommend-web start\` → \`/status\` returns \`event_store_ready: true\` after setup
- [ ] Remove zip files, run \`./recommend "spy thriller"\` → ordinary recommendations still work
- [ ] \`/logs?file=web\` shows \`web.log\` content; Clear truncates only the selected file; switching files preserves line count

### CODE REVIEW

**Review by achandra-rp — COMMENTED** [review]
Starting review for PR 29.

File: recommender/event_store.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+            conn.execute("DELETE FROM imports WHERE provider = ?", (provider,))
+
+            # Insert new import row
+            cursor = conn.execute(
+                "INSERT INTO imports (provider, source_path, source_sha256) VALUES (?, ?, ?)",
+                (provider, source_path, source_sha256),
+            )
+            import_id = cursor.lastrowid
+
+            # Insert events with dedup via INSERT OR IGNORE on source_hash
```
Comment (achandra-rp) [inline recommender/event_store.py:110]: <COMMENT>
[LOW] For large watch history imports, executing `conn.execute` in a loop can be relatively slow. Using `conn.executemany` with a batch of pre-computed event values would significantly improve performance for bulk insertions.

```suggestion
            # Insert events with dedup via INSERT OR IGNORE on source_hash
            events_data = []
            for event in events:
                ts_iso = event.timestamp.isoformat(timespec="seconds")
                duration_secs = int(event.watched_duration.total_seconds())
                total_secs = (
                    int(event.total_duration.total_seconds())
                    if event.total_duration is not None
                    else None
                )
                source_hash = _compute_source_hash(
                    provider, event.profile, event.title,
                    ts_iso, duration_secs,
                )
                events_data.append((
                    provider, event.title, event.content_type, event.series_name,
                    duration_secs, total_secs, ts_iso, event.profile,
                    import_id, source_hash
                ))

            conn.executemany(
                "INSERT OR IGNORE INTO watch_events "
                "(provider, title, content_type, series_name, "
                "watched_duration_seconds, total_duration_seconds, "
                "timestamp_iso, profile, import_id, source_hash) "
                "VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)",
                events_data
            )
```
</COMMENT>

**Review by achandra-rp — COMMENTED** [review]
## 📋 Review Summary

The PR successfully implements an event store using SQLite to persist normalized watch history, significantly improving application startup time by avoiding repetitive zip file parsing on every load. The introduction of lazy loading in `RecommendContext` is a clean architectural improvement that ensures database access only occurs when watch history events are actually queried.

## 🔍 General Feedback

- The move to SQLite for persistence is a great optimization for the application's scalability and responsiveness.
- The lazy loading pattern in `RecommendContext` is well-implemented and preserves backward compatibility with eager event lists.
- Test coverage for the new `event_store` module is comprehensive and addresses key edge cases like deduplication and provider isolation.
- Consider using `executemany` in `event_store.py` to further optimize bulk insertions for very large watch histories.
